### ISSUE TITLE
Npapi - Adobe Flash player - Waterfox Classic 2020.12

### ISSUE THREAD

**Issue by Lins31** [MrAlex94/Waterfox#1916]
How to I install Npapi - Adobe Flash player - libflashplayer.so
It keeps getting blocked by Firefox

Im on Waterfox Classic 2020.12 - and Running Linux Mint 19.3

**Comment by Jamesstevens-3rd** [comment]
I can't help with the actual installing of applications on Linux but this [Adobe Flash Player 32.0.0.371](http://web.archive.org/web/20200630185943if_/https://fpdownload.macromedia.com/pub/flashplayer/installers/archive/fp_32.0.0.371_archive.zip) this archive has Windows, MAC and Linux versions for IE, Firefox etc. 

**NOTE**: **_32.0.0.371_** is the last version of Flash Player without the _Time-bomb_

TIP: I would advise you install an Addon that allows you to enable/disable the plugin at will like QuickJava

http://quickjavaplugin.blogspot.com/p/welcome.html
https://github.com/veo/Firefox-addons/raw/master/quickjava-2.1.2-fx.xpi

**Comment by magiruuvelvet** [comment]
You know Flash is dead as s\*\*\* and unsupported literally everywhere. Why would you want to use Flash in 2021? (in your browser??)


https://www.adobe.com/products/flashplayer/end-of-life.html

> Since Adobe no longer supports Flash Player after December 31, 2020 and blocked Flash content from running in Flash Player beginning January 12, 2021, Adobe strongly recommends all users immediately uninstall Flash Player to help protect their systems.

**Comment by Jamesstevens-3rd** [comment]
>Why would you want to use Flash in 2021? (in your browser??)

for all those sites that still use it I suppose, just because you cant use it doesn't mean other people don't have a use for it, personally I have little use for flash player and if you are so concerned about outdated software why are you posting a comment in a browser based on Firefox 56 (2017)

if you search for flash on the Waterfox subreddit you will see lost of posts about this.
https://www.reddit.com/r/waterfox/

**Comment by Lins31** [comment]
Why would I want to still use Flash - I play a community game - https://s1.miramagia.com/game.php
It uses Flash so we can interact and communicate in many different ways - these functions are not yet available in HTML5
Where else can you turn someone into a Unicorn and chase them as a Polar Bear - while chatting
Im tasked with looking for solutions for our game. 

Anyway - I found the root of my problem.  **Firefox updates**  Firefox is updating x2 a week! 
It deletes / replaces my local Mozilla file (Linux - .mozilla)
This makes Firefox disfuctional as a browser -as it deletes all extensions / plugins / history /bookmarks /search engine prefs etc 
It also affects Waterfox

**This issue is not closed**
Next we have to get Apple Mac players online

**Comment by Jamesstevens-3rd** [comment]
>This makes Firefox disfuctional as a browser -as it deletes all extensions / plugins / history /bookmarks /search engine prefs etc
>It also affects Waterfox

this will not affect Waterfox Classic, as its based on a version of Firefox 56 and will never go to Quantum, Waterfox Classic will never update to G3, however Firefox on the other hand will always update itself to the latest.

**Comment by Lins31** [comment]
**Im sorry it does affect Waterfox Classic.**  Flash is not embedded in Firefox - and it draws the extension through .mozilla 
the Flash error msg is inconsistant (#2046 #2032 #4056 #5731) - but it stops flash from working :-(

Solution - to delete .mozilla (as Firefox updates have already done this) before running Waterfox Classic

**Comment by MrAlex94** [comment]
Sorry, but NPAPI support will remain as is. It's already a large attack vector, I don't have the resources to implement any changes or fixes.

**Comment by DraconicNEO** [comment]
> You know Flash is dead as s*** and unsupported literally everywhere. Why would you want to use Flash in 2021? (in your browser??)
> 
> https://www.adobe.com/products/flashplayer/end-of-life.html
> 
> > Since Adobe no longer supports Flash Player after December 31, 2020 and blocked Flash content from running in Flash Player beginning January 12, 2021, Adobe strongly recommends all users immediately uninstall Flash Player to help protect their systems.

This is a browser dedicated for Legacy support.
Also there are newer versions of Flash than version 32 that still get updates (at the time of writting this) see [cleanflash](https://github.com/cleanflash/installer) and [Flashpatch](https://github.com/darktohka/FlashPatch) for more info on this.

**Comment by Dwedit** [comment]
SRWare Iron 84.0.4300.0 will run Flash (assuming you have disabled the time bomb), though there are a few steps to get it enabled, then it auto-disables on each new page until you explicitly allow it.

MSIE will also run Flash.
